pyp2spec/pyp2spec.py

```diff
@@ -14,10 +14,16 @@
     "--spec-output", "-o",
     help="Provide custom output where spec file will be saved",
 )
+@click.option(
+    "--declarative-buildsystem", is_flag=True, default=False,
+    help="Create a spec file with pyproject declarative buildsystem (experimental)",
+)
 def main(**options):  # noqa
     try:
+        if options["automode"] and options["declarative_buildsystem"]:
+            raise Pyp2specError("Declarative buildsystem doesn't work with automode")
         config_file = create_config(options)
-        create_spec_file(config_file, options["spec_output"])
+        create_spec_file(config_file, options["spec_output"], options["declarative_buildsystem"])
```

Comment:
For example here, `create_spec_file` could take the entire `options`  dict.

Reply:
It could: https://github.com/befeleme/pyp2spec/pull/55/commits/ded9746d7e30a820f3a590d545b946b6adc92eda